src/component/base/DraftEditor.react.js

```diff
@@ -354,13 +354,16 @@ class DraftEditor extends React.Component<DraftEditorProps, State> {
    *
    * We attempt to preserve scroll position when focusing. You can also pass
    * a specified scroll position (for cases like `cut` behavior where it should
-   * be restored to a known position).
+   * be restored to a known position).  In situations where you need to also update
+   * EditorState and focus, you can pass your new EditorState as a param to avoid race conditions.
    */
-  focus = (scrollPosition?: DraftScrollPosition): void => {
-    const {editorState} = this.props;
+  focus = (
+    scrollPosition?: DraftScrollPosition,
+    newEditorState?: EditorState,
```

Comment:
I would like to think of other alternatives to solve this problem instead of just passing a new editorState as an argument.